processor/src/chiplets/hasher/mod.rs

```diff
@@ -230,7 +270,14 @@ impl Hasher {
             //   operation batch on the last row.
             // - last permutation: continue hashing on the first row, and return the result
             //   on the last row.
-            self.trace.append_permutation(&mut state, START, ABSORB);
+            if is_memoized {
+                // copy trace if trace exists
+                self.trace.copy_trace(&mut state, row..(row + 8));
+                row += 8;
```

Comment:
We should use `HASH_CYCLE_LEN` here and in all cases, instead of the magic number 8

Reply:
Fixed